### PR TITLE
Add win_psrepository_info module

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY

Adding new module `win_psrepository_info` to provide information on PSRepositories.

Besides being able to just generally get this info, it allows for some manipulation of repositories based on their current settings, such as this example from the docs:

```yaml
- name: Get info for all repositories
  win_psrepository_info:
  register: repo_info

- name: Remove all repositories that don't have a publish_location set
  win_psrepository:
    name: "{{ item }}"
    state: absent
  loop: "{{ repo_info.repositories | rejectattr('publish_location', 'none') | list }}"
```

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

win_psrepository_info

##### ADDITIONAL INFORMATION 
I added a function to this that does basically the same thing as `Convert-DictToSnakeCase` but for objects (using their properties). 

Fine to keep it contained in this module but I'm also happy to move that into `Ansible.ModuleUtils.CamelConversion` if that's desirable (would rather do that in a new PR that removes it from here at the same time); I think it'd be useful in other modules. 

I'd personally use it right away for a `win_psmodule_info` and a `win_psscript_info` module for instance.

##### RELATED PRs
- #67499 
- #67674

### CODE REVIEW

File: lib/ansible/modules/windows/win_psrepository_info.ps1 [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!powershell
+
+# Copyright: (c) 2020, Brian Scholer <@briantist>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#Requires -Module Ansible.ModuleUtils.CamelConversion
+#Requires -Module PowerShellGet
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_psrepository_info.ps1:8]: This has no effect when running through Ansible. The way we run modules in it's own pipeline means these `#Requires` checks are skipped.

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
Looks good, just added a comment but that doesn't need to change, just an FYI.

File: lib/ansible/modules/windows/win_psrepository_info.ps1 (hunk 1 above)
Comment (briantist) [reply lib/ansible/modules/windows/win_psrepository_info.ps1:8]: Will keep it in mind for the future, thanks!

**Comment by briantist** [comment]
> Looks good, just added a comment but that doesn't need to change, just an FYI.
Even though Ansible doesn't use the `#Requires` statement, do you have an objection to keeping them/adding more in other modules? It might not be used now but could become useful later, as it's much easier to determine required modules through static analysis with those.

**Comment by jborean93** [comment]
That's a good point, we could potentially have a future sanity check to make sure any 3rd party modules that are documented with `#Requires` are also documented in the docs for that module properly.

**Pull request merged**